Models/Board.cs

```diff
@@ -37,11 +37,46 @@ public Board(List<Piece> pieces)
 
         public bool IsTileOccupied(Tile tile)
         {
-            return Position.ContainsKey(tile);
+            return Positions.ContainsKey(tile);
         }
         public bool IsTileOccupied(int row, int column)
         {
-            return Position.ContainsKey(new Tile(row, column));
+            return Positions.ContainsKey(new Tile(row, column));
+        }
+
+        public void ApplyMove(Movement move)
```

Comment:
The board shouldn't be responsible for making the movement, as we said before, the movement will be represented with a new board created for each movement, so this method should be removed(ApplyMove)